pkg/kubectl/delete.go

```diff
@@ -368,7 +368,7 @@ func (reaper *JobReaper) Stop(namespace, name string, timeout time.Duration, gra
 	// TODO: handle overlapping jobs
 	retry := NewRetryParams(reaper.pollInterval, reaper.timeout)
 	waitForJobs := NewRetryParams(reaper.pollInterval, timeout)
-	if err = scaler.Scale(namespace, name, 0, nil, retry, waitForJobs); err != nil {
+	if err = scaler.Scale(namespace, name, 0, nil, retry, waitForJobs); err != nil && !errors.IsNotFound(err) {
```

Comment:
I think this PR should focus on fixing the flake instead of refactoring Scaler. 

Would you keep the Scaler as-is, and check these returned error the way `DeploymentReaper` does, to see if it fixes the flake?
```go
iscaleGetErr, ok := err.(ScaleError)
if errors.IsNotFound(err) || (ok && errors.IsNotFound(scaleGetErr.ActualError)) {
// ...
```

Making the scale client to return the actual API error can be done in another PR.

Reply:
> Would you keep the Scaler as-is, and check these returned error the way DeploymentReaper does, to see if it fixes the flake?

It's actually pretty hard to reason about which places get "normal" errors and which don't.  I ended up doing this after messed up the change a couple times.



> Did #60455 cause the flake? If so, why?

I suspect it originated at the scale client errors instead of the non-scale client errors.

Reply:
> It's actually pretty hard to reason about which places get "normal" errors and which don't. I ended up doing this after messed up the change a couple times.

Did you mean the returned error could be a 404, but is neither `errors.IsNotFound` nor `ScaleError`?

Reply:
> Did you mean the returned error could be a 404, but is neither errors.IsNotFound nor ScaleError?

Yeah.  It was when I started and I started peeling back layers until I found the scale client.